addons/analytic/static/src/components/analytic_distribution/analytic_distribution.js

```diff
@@ -72,23 +78,28 @@ export class AnalyticDistribution extends Component {
         // and thus different applicabilities apply
         // or a model applies that contains unavailable plans
         // This should only execute when these fields have changed, therefore we use the `_field` props.
-        // (consider including the plans in the computed json, python side)
         const valueChanged = JSON.stringify(this.props.value) !== JSON.stringify(nextProps.value);
-        if (this.applicabilityParamsChanged(nextProps) || valueChanged) {
-            await this.fetchAllPlans(nextProps);
-            await this.formatData(nextProps);
+        if (valueChanged) {
+            if (this.props.force_applicability) {
+                this.formatData(nextProps);
```

Comment:
Seems like you always need to format the data, so why don't you always call it at the end of this function?
And remove the function `fetchPlansAndFormatData` since you can only call `fetchAllPlans` when needed

Reply:
This function executes for every prop change, for example clicking on the row, or changing the price. It is already executing more than necessary. Value changed manually or by a model is interpreted as the same and calls `formatData` for both cases when it is only required by the latter.

But I get your point...

Reply:
I don't think that what I'm suggesting changes anything regarding the number of calls of the functions

Reply:
improved the willUpdate function